multimap/src/main/java/org/infinispan/multimap/api/MultimapCache.java

```diff
@@ -0,0 +1,226 @@
+ * <p>
+ * <h2>Example</h2>
+ * <pre>
+ *
+ *    multimapCache.put("k", "v1").join();
+ *    multimapCache.put("k", "v2").join();
+ *    multimapCache.put("k", "v3").join();
+ *
+ *    Collection<String> results = multimapCache.get("k").join();
+ *
+ * </pre>
+ * <p>
+ * <h2>Eviction</h2> <p> Eviction works per key. This means all the values associated on a key will be evicted. </p>
+ * <p>
+ * <h2>Views</h2>
+ * <p>
+ * The returned collections when calling "get" are views of the values on the key. Any change on these collections won't
+ * affect the cache values on the key.
+ * <p>
+ * <h2>Null values</h2> Null values are not supported. The multimap cache won't have a null key or any null value.
+ * <p>
+ * Example
+ * <pre>
+ *     multimapCache.put(null, "v1").join() -> fails
+ *     multimapCache.put("k", null).join() -> fails
+ *     multimapCache.put("k", "v1").join() -> works and add's v1
+ *     multimapCache.containsKey("k").join() -> true
+ *     multimapCache.remove("k", "v1").join() -> works, removes v1 and as the remaining collection is empty, the key is
+ * removed
+ *     multimapCache.containsKey("k").join() -> false
+ *  </pre>
+ * <p>
+ * <p>
+ * <h2>Duplicates</h2> Today's implementation does not support duplicate values on keys. {@link Object#equals(Object)}
+ * method is used to check if a value is already present in the key. This means that the following code
+ * <pre>
+ *    multimapCache.put("k", "v1").join();
+ *    multimapCache.put("k", "v2").join();
+ *    multimapCache.put("k", "v2").join();
+ *    multimapCache.put("k", "v2").join();
+ *
+ *    multimapCache.get("k").thenAccept(values -> System.out.println(values.size()));
+ * </pre>
+ * prints the value 2. "k" -> ["v1", "v"]
+ * <p>
+ * Future implementations might evolve to support duplicated values.
+ * <p>
+ * <h2>Transactions</h2> MultimapCache supports implicit transactions without blocking. The following methods block when
+ * they are called in a explicit transaction context. This limitation could be improved in the followin versions if
+ * technically possible. <ul> <li>{@link MultimapCache#size()}</li> <li>{@link MultimapCache#containsEntry(Object,
+ * Object)}</li> <li>{@link MultimapCache#remove(Predicate)}</li> <li></li> </ul> More about transactions in : <a
+ * href="http://infinispan.org/docs/dev/user_guide/user_guide.html#transactions">the Infinispan Documentation</a>.
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @see <a href="http://infinispan.org/documentation/">Infinispan documentation</a>
+ * @since 9.2
+ */
+@Experimental
+public interface MultimapCache<K, V> {
+
+   /**
+    * Asynchronous method. Puts a key-value pair in this multimap cache.
+    * <ul>
+    * <li>If this multimap cache supports duplicates, the value will be always added.</li>
+    * <li>If this multimap cache does <i>not support</i> duplicates and the value exists on the key,
+    * nothing will be done.</li>
+    * </ul>
+    *
+    * @param key,   the key to be put
```

Comment:
why not, comma are pretty 💃 
TBH, I don't know why or when I took the idea from, and I repeated the , in my javadoc. Removing them